automata/fa/dfa.py

```diff
@@ -162,3 +163,12 @@ def _init_from_nfa(self, nfa):
             states=dfa_states, input_symbols=dfa_symbols,
             transitions=dfa_transitions, initial_state=dfa_initial_state,
             final_states=dfa_final_states)
+
+    def num_accepted(self, length):
```

Comment:
This function would be better served returning a generator instead of the full list. In practice, that means you would `yield w` within your loop instead of building up the `accepted` list of results. In this case, you would still have access to the list of accepted permutations, and if you need the length, you could `len(list(dfa.num_accepted(length=3)))`